src/Tap.sol

```diff
@@ -75,21 +87,27 @@ contract Tap {
         require(now == rho, "Tap/rho-not-updated");
         if (what == "flight") flight = data;
         else revert("Tap/file-unrecognized-param");
+
+        emit File(what, data);
     }
     function file(bytes32 what, uint256 data) external auth {
         require(now == rho, "Tap/rho-not-updated");
         if (what == "rate") rate = data;
         else revert("Tap/file-unrecognized-param");
+
+        emit File(what, data);
     }
 
+    // --- External ---
     function pump() external stoppable {
-        require(now >= rho, "Tap/invalid-now");
-        uint256 wad = mul(now - rho, rate);
-        if (wad > 0) {
-            vat.suck(address(vow), address(this), mul(wad, RAY));
-            daiJoin.exit(address(this), wad);
-            keg.pour(flight, wad);
-        }
+        require(now > rho, "Tap/invalid-now");
+        uint256 wad = mul(now - rho, rate);  // TODO How is that working? We clearly need to increase the test coverage
```

Comment:
Just noticed this, but `now` should be changed to `block.timestamp`. `now` is deprecated.

What's the issue with the second line there? `rate = DAI / second in WADs` so `TIME ELAPSED IN SECONDS * RATE = DAI to pay in WADs`.

Reply:
Oh got it sorry, is dai/sec a good metric?

Reply:
It's definitely the most natural as blockchain time is in seconds.

Reply:
I would have used the natural business one. I set the allowance at 400000 / year for .... 
However I have removed the comment.